### PR TITLE
Add bobcat 2023.2/candidate channels

### PR DESCRIPTION
**Pull request by ajkavanagh**
Add the bobcat 2023.2 channes for the OpenStack charms. The recipe is to
push to /candidate for SQA to consume prior to releasing to stable. Bug
fixes during the testing period will therefore be pushed to /candidate
to assist with simplified testing of any fixes.

The charmcraft 2.2/stable channel has been selected for the recipes as
that is the latest stable channel availble for long term building of the
bobcat charms.

Bases 22.04 and 23.10 have been selected to support jammy and mantic.

### CODE REVIEW

File: charmed_openstack_info/data/lp-builder-config/openstack.yaml
Comment (freyes) [inline charmed_openstack_info/data/lp-builder-config/openstack.yaml:97]: Blocking until https://github.com/canonical/craft-providers/pull/330 gets available in a release, or we agree something different.

**Review by freyes — CHANGES_REQUESTED** [review]

File: charmed_openstack_info/data/lp-builder-config/openstack.yaml
Comment (fnordahl) [inline charmed_openstack_info/data/lp-builder-config/openstack.yaml:2001]: A rebase is needed after #40 merged, it provided a commented out section for the stable channel, which I guess can be uncommented as part of this PR

**Comment by freyes** [comment]
Blocked by https://github.com/canonical/charmcraft/pull/1354

**Review by freyes — APPROVED** [review]

**Pull request merged**
